kustomize/commands/internal/util/configmapSecretFlagsAndArgs.go

```diff
@@ -62,32 +68,49 @@ func (a *ConfigMapSecretFlagsAndArgs) Validate(args []string) error {
 	return nil
 }
 
+// ValidateSet validates required fields are set to support structured generation for the
+// edit set command.
+func (a *ConfigMapSecretFlagsAndArgs) ValidateSet(args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("name must be specified once")
+	}
+	a.Name = args[0]
+
+	if len(a.LiteralSources) == 0 && a.NewNamespace == "" {
+		return fmt.Errorf("at least one of [--from-literal, --new-namespace] must be specified")
+	}
+
+	return nil
+}
+
 // ExpandFileSource normalizes a string list, possibly
 // containing globs, into a validated, globless list.
 // For example, this list:
-//     some/path
-//     some/dir/a*
-//     bfile=some/dir/b*
+// - some/path
+// - some/dir/a*
+// - bfile=some/dir/b*
 // becomes:
-//     some/path
-//     some/dir/airplane
-//     some/dir/ant
-//     some/dir/apple
-//     bfile=some/dir/banana
+// - some/path
+// - some/dir/airplane
+// - some/dir/ant
+// - some/dir/apple
+// - bfile=some/dir/banana
 // i.e. everything is converted to a key=value pair,
 // where the value is always a relative file path,
 // and the key, if missing, is the same as the value.
 // In the case where the key is explicitly declared,
 // the globbing, if present, must have exactly one match.
 func (a *ConfigMapSecretFlagsAndArgs) ExpandFileSource(fSys filesys.FileSystem) error {
+	const NumberComponentsWithKey = 2
```

Comment:
Nit:
```suggestion
	const numberComponentsWithKey = 2
```
Shouldn't matter since this is a local var, but we need not export it.

Reply:
I intended this to be a constant, since it used to be a magic number before (and the linter was complaining 😅). I agree with not exporting, but changing this to a var can be problematic if any future changes accidentally modify this value.

Reply:
Updated!